src/sagemaker/jumpstart/types.py

```diff
@@ -794,26 +858,26 @@ class JumpStartModelSpecs(JumpStartDataHolderType):
         "model_subscription_link",
     ]
 
-    def __init__(self, spec: Dict[str, Any]):
-        """Initializes a JumpStartModelSpecs object from its json representation.
+    def __init__(self, fields: Optional[Dict[str, Any]]):
+        """Initializes a JumpStartMetadataFields object.
 
         Args:
-            spec (Dict[str, Any]): Dictionary representation of spec.
+            fields (Dict[str, Any]): Dictionary representation of metadata fields.
         """
-        self.from_json(spec)
+        self.from_json(fields)
```

Comment:
how does this work, I don't `mypy` should accept this signature given that `from_json` accepts a `Dict[str, Any]` and get passed an `Optional[Dict[str, Any]]`. This could cause null pointer errors.

Reply:
Sorry this is confusing typing, the `fields` object is `Dict[str, Any]` but not optional. I think `mypy` is not checking these typings, updated in my next PR